transform/instance.go

```diff
@@ -344,3 +444,39 @@ func (st *scalarTransformer) transform(in interface{}, modifier pathModifier) (i
 	// 3. Fall back to the JSON Schema default value.
 	return st.defaultValue, nil
 }
+
+// transformScalarXML retrieves the value for a scalar instance following this process:
+//
+// 1. Use a Transform if it exists.
+//
+// 2. Fall back to the JSON Schema default value.
+func (st *scalarTransformer) transformScalarXML(in interface{}, modifier pathModifier, transformType string) (interface{}, error) {
+	path := st.jsonPath
+	if modifier != nil {
+		path = modifier(path)
+	}
+	// 1. Use a transform if it exists
+	if st.transforms != nil {
+		newValue, err := st.transforms.transform(in, st.jsonType, modifier, transformType)
+		if err != nil {
+			return nil, err
+		}
+		if newValue != nil {
+			return newValue, nil
+		}
+	}
+
+	// 2. Fall back to the JSON Schema default value.
+	return st.defaultValue, nil
+}
+
+// transform routes to the correct scalar transform type
+func (st *scalarTransformer) transform(in interface{}, modifier pathModifier, transformType string) (interface{}, error) {
+	if transformType == "json" {
```

Comment:
This is what I was thinking but you had to change transform command to get this to work by adding another argument. I thought it may be simpler to just set the transform type in the newScalarTransform function. Then here it is just like this but `if st.transformType == "json"`